src/backend/distributed/executor/multi_utility.c

```diff
@@ -835,6 +857,53 @@ ErrorIfDistributedRenameStmt(RenameStmt *renameStatement)
 
 
 /*
+ * CreateLocalTable gets DDL commands from the remote node for the given
+ * relation. Then, it creates the local relation as temporary and on commit drop.
+ */
+static void
+CreateLocalTable(char *relationName, char *nodeName, int32 nodePort)
+{
+	List *ddlCommandList = NIL;
+	ListCell *ddlCommandCell = NULL;
+
+	/* fetch the ddl commands needed to create the table */
+	StringInfo tableNameStringInfo = makeStringInfo();
+	appendStringInfoString(tableNameStringInfo, relationName);
+
+	/*
+	 * The warning message created in TableDDLCommandList() is descriptive
+	 * enough; therefore, we just throw an error which says that we could not
+	 * run the copy operation.
+	 */
+	ddlCommandList = TableDDLCommandList(nodeName, nodePort, tableNameStringInfo);
+	if (ddlCommandList == NIL)
+	{
+		ereport(ERROR, (errmsg("could not run copy from the worker node")));
+	}
+
+	/* apply DDL commands against the local database */
+	foreach(ddlCommandCell, ddlCommandList)
+	{
+		StringInfo ddlCommand = (StringInfo) lfirst(ddlCommandCell);
+		Node *ddlCommandNode = ParseTreeNode(ddlCommand->data);
+
+		if (IsA(ddlCommandNode, CreateStmt))
+		{
+			CreateStmt *createStatement = (CreateStmt *) ddlCommandNode;
+
+			/* create the local relation as temporary and on commit drop */
+			createStatement->relation->relpersistence = RELPERSISTENCE_TEMP;
+			createStatement->oncommit = ONCOMMIT_DROP;
+		}
+
+		ProcessUtility(ddlCommandNode, ddlCommand->data, PROCESS_UTILITY_TOPLEVEL,
```

Comment:
For the server, I think you could do something like:

        else if (IsA(ddlCommandNode, CreateForeignServerStmt))
        {
            CreateForeignServerStmt *createServerStmt = (CreateForeignServerStmt *) ddlCommandNode;
            if (GetForeignServerByName(createServerStmt->servername, true) != NULL)
            {
                /* server already exists, silently skip */
                continue;
            }
        }

There are some other issues with cstore_fdw. Since foreign tables are not temporary, the table still exists after the COPY command and at that point cstore_fdw intercepts the COPY.